clinica/pydra/t1_volume/dartel2mni/pipeline.py

```diff
@@ -0,0 +1,158 @@
+from typing import Any
+
+import pydra
+from nipype.algorithms.misc import Gunzip
+from nipype.interfaces.spm import DARTELNorm2MNI
+from pydra import Workflow
+from pydra.tasks.nipype1.utils import Nipype1Task
+
+from clinica.pydra.engine import clinica_io
+
+
+def _check_pipeline_parameters(parameters: dict) -> dict:
+    """Check the parameters passed to the pipeline.
+
+    Parameters
+    ----------
+    parameters : dict
+        Dictionary of parameters to analyze.
+
+    Returns
+    -------
+    dict :
+        Cleaned dictionary of parameters.
+    """
+    from clinica.utils.group import check_group_label
+
+    if "group_label" not in parameters:
+        raise KeyError("Missing compulsory group_label key in pipeline parameter.")
+    check_group_label(parameters["group_label"])
+    parameters.setdefault("tissues", [1, 2, 3])
+    parameters.setdefault("voxel_size", None)
+    parameters.setdefault("modulate", True)
+    parameters.setdefault("smooth", 8.0)
+
+    return parameters
+
+
+@clinica_io
+def build_core_workflow(
+    name: str = "t1-volume-dartel2mni", parameters: dict = {}
```

Comment:
I agree with you that we should definitely get rid of this pattern which basically implements the sanitization and default value twice. It is error prone and it has already shown that it could be a real pain point with FWHM values for example.
But, I also think having a consistent api across pipelines for `build_core_workflow` is very valuable. Atm, we have the implicit convention 

`build_core_workflow(name: str, parameters: dict) -> Workflow`  

where parameters is a simple dictionary of user-provided parameters through the CLI. Since the CLI parametrization already provides some checks (types, existence of paths, default values...), maybe it'd make sense to implement all the sanitization of the parameters before passing them to `build_core_workflow` ? 🤔

Reply:
See [d5ef4c1](https://github.com/aramis-lab/clinica/pull/815/commits/d5ef4c1b881ea37fde05198974ac898de3fcccc9)